src/inference/src/cpp_interfaces/interface/ie_iinfer_request_internal.cpp

```diff
@@ -103,6 +103,8 @@ void IInferRequestInternal::SetBlob(const std::string& name, const Blob::Ptr& us
         !((foundInput && foundInput->getInputData()->isDynamic()) || (foundOutput && foundOutput->isDynamic()))) {
         IE_THROW() << "Input data is empty. Input name: \'" << name << "\'";
     }
+    const bool isInputDynamic = foundInput && foundInput->getInputData()->isDynamic();
```

Comment:
this API is deprecated https://github.com/openvinotoolkit/openvino/blob/master/src/inference/include/ie/ie_data.h#L184-L185

BTW, why AUTO even need to call these methods? It should work with device-specific SetBlob / GetBlob like HETERO does.

See dynamism support in HETERO https://github.com/openvinotoolkit/openvino/pull/8012/files

CC @apankratovantonp

Reply:
Hi ilya, per my understanding, For hetero, each SetBlob/GetBlob of hetero infer request can be forwarded to sub-requests, which finally run on device-specific SetBlob/GetBlob. But for AUTO/Mutil, this is not the case. in auto/multi, the actual device infer request is been scheduled when doing stage1/stage2 of auto/multi async pipeline, which means when setting/getting blob, auto/multi does not know which actual device infer request will be scheduled for this auto/multi infer request. that's why I need to make change here to make auto work with dynamic shape.

Reply:
right, Auto/Multi have to keep it's own blobs to be able to Set them to the actual device upon selection